cmd/glasskube/cmd/purge.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/glasskube/glasskube/internal/clicontext"
+	"github.com/glasskube/glasskube/internal/cliutils"
+	"github.com/glasskube/glasskube/internal/config"
+	"github.com/glasskube/glasskube/pkg/bootstrap"
+	"github.com/glasskube/glasskube/pkg/purge"
+	"github.com/glasskube/glasskube/pkg/statuswriter"
+	"github.com/spf13/cobra"
+)
+
+type purgeOptions struct {
+	yes bool
+}
+
+var purgeCmdOptions = purgeOptions{}
+
+var purgeCmd = &cobra.Command{
+	Use:   "purge",
+	Short: "Purge Glasskube from a Kubernetes cluster",
+	Long: "Purges Glasskube from a Kubernetes cluster, " +
+		"thereby uninstalling the Glasskube operator and all related resources.",
+	Args:   cobra.NoArgs,
+	PreRun: cliutils.SetupClientContext(false, &rootCmdOptions.SkipUpdateCheck),
+	Run: func(cmd *cobra.Command, args []string) {
+		cfg, _ := cliutils.RequireConfig(config.Kubeconfig)
+		client := purge.NewPurger(cfg)
+		ctx := cmd.Context()
+
+		if !rootCmdOptions.NoProgress {
+			client.WithStatusWriter(statuswriter.Spinner())
+		}
+
+		bold := color.New(color.Bold).SprintFunc()
+		currentContext := clicontext.RawConfigFromContext(ctx).CurrentContext
+
+		if !purgeCmdOptions.yes {
+			confirmMessage := fmt.Sprintf("⚠️ Glasskube and all related resources will be purged from context %s."+
+				"\nThis includes removal of all installed packages!\nContinue? ", bold(currentContext))
+			if !cliutils.YesNoPrompt(confirmMessage, true) {
+				fmt.Fprintln(os.Stderr, "Operation stopped")
+				cliutils.ExitWithError()
+			}
+		}
+
+		IsBootstrapped, err := bootstrap.IsBootstrapped(cmd.Context(), cfg)
+		if err != nil && !IsBootstrapped {
+			fmt.Fprintf(os.Stderr, "error: %v\n", err)
+			cliutils.ExitWithError()
+		}
```

Comment:
This doesn't work. You have to check `isBootstrapped` (please change to lower-case first letter) and `err` separately. `err` indicates if there was an error checking the bootstrap status and if `err` is `nil`, `isBootstrapped` indicates whether the cluster has been bootstrapped or not.

Also, I think it would be better to perform this check before the confirmation prompt.